libsolidity/parsing/Parser.cpp

```diff
@@ -599,13 +607,17 @@ ASTPointer<ASTNode> Parser::parseFunctionDefinition(bool _freeFunction)
 			m_scanner->next();
 		}
 		else
+		{
+			nameLocation = currentLocation();
 			name = expectIdentifierToken();
+		}
 	}
 	else
 	{
 		solAssert(kind == Token::Constructor || kind == Token::Fallback || kind == Token::Receive, "");
 		m_scanner->next();
 		name = make_shared<ASTString>();
+		nameLocation = currentLocation();
```

Comment:
I think this has to be empty here - the only other option would be to be around the keyword (i.e. above the call to `next`), but I'm not sure that will work.